### ISSUE TITLE
L.Control.Geocoder.HERE fails to parse JSON (Firefox)

### ISSUE THREAD

**Issue by sreejeshmail** [perliedman/leaflet-control-geocoder#154]
I have a map search box in my application. I am using here map . But the search is not working in firefox , 
google chrome it is working. Is there any specific reason why it is not working with firefox ? 

This is the search code:

```
 L.Control.Geocoder.getJSON = function(url, params, callback) {
            	    var xmlHttp = new XMLHttpRequest();
            		xmlHttp.open( "GET", url + L.Util.getParamString(params), true);
               	xmlHttp.send(null);
            		xmlHttp.onreadystatechange = function () {
            			if (xmlHttp.readyState != 4) return;
            			if (xmlHttp.status != 200 && req.status != 304) return;
            			callback(JSON.parse(xmlHttp.response));
            			
            		};
            
          	};
```

**Comment by sreejeshmail** [comment]
Any suggestion ?

**Comment by perliedman** [comment]
Can you provide an error message or similar, or preferably a running example of the issue with jsfiddle or similar?

I can see that this geocoder example is working in Firefox: http://www.liedman.net/leaflet-control-geocoder/

**Comment by sreejeshmail** [comment]
![firefox-error](https://cloud.githubusercontent.com/assets/23445401/20293546/7491caa6-ab1d-11e6-82cc-48062e9becf6.jpg)

please see the screen shot

**Comment by sreejeshmail** [comment]
search is working in firefox after modifying the function like this 

 L.Control.Geocoder.getJSON = function(url, params, callback) {
            	    var xmlHttp = new XMLHttpRequest();
            		xmlHttp.open( "GET", url + L.Util.getParamString(params), true);
            		xmlHttp.setRequestHeader("Accept", "application/json");
            		xmlHttp.send(null);
            		xmlHttp.onreadystatechange = function () {
            			if (xmlHttp.readyState != 4) return;
            			if (xmlHttp.status != 200 && req.status != 304) return;
            			callback(JSON.parse(xmlHttp.response));
            			
            		};
            
          	};

i added   	xmlHttp.setRequestHeader("Accept", "application/json");

**Comment by sreejeshmail** [comment]
is that right way  ?

**Comment by sreejeshmail** [comment]
we are using 1.0.0 version ..

**Comment by sreejeshmail** [comment]
![ff-error](https://cloud.githubusercontent.com/assets/23445401/20295082/dc835f70-ab28-11e6-8428-695a54d12015.jpg)

This is the error i m getting

**Comment by simon04** [comment]
Fixed in f0827ecc6172578e1f22834bdd65164636f26a81.
